core/templates/services/playthrough.service.ts

```diff
@@ -184,28 +183,47 @@ export class PlaythroughService {
               PlaythroughService.multipleIncorrectStateName.num_times_incorrect)
           }
         };
+
+        PlaythroughService.playthrough = (
+          this.playthroughObjectFactory
+            .createNewMultipleIncorrectSubmissionsPlaythrough(
+              currentPlaythrough.playthroughId, currentPlaythrough.expId,
+              currentPlaythrough.expVersion, issueCustomizationArgs,
+              currentPlaythrough.actions));
       } else if (this.isCyclicStateTransitionsIssue()) {
-        PlaythroughService.playthrough.issueType = (
-          AppConstants.ISSUE_TYPE_CYCLIC_STATE_TRANSITIONS);
-        PlaythroughService.playthrough.issueCustomizationArgs = {
+        const currentPlaythrough = PlaythroughService.playthrough;
+        const issueCustomizationArgs = {
           state_names: {
             value: PlaythroughService.cycleIdentifier.cycle.split(',')
           }
         };
+
+        PlaythroughService.playthrough = (
+          this.playthroughObjectFactory
+            .createNewCyclicStateTransitionsPlaythrough(
+              currentPlaythrough.playthroughId, currentPlaythrough.expId,
+              currentPlaythrough.expVersion, issueCustomizationArgs,
+              currentPlaythrough.actions));
       } else if (this.isEarlyQuitIssue(timeSpentInExpInSecs)) {
-        PlaythroughService.playthrough.issueType = (
-          AppConstants.ISSUE_TYPE_EARLY_QUIT);
-        PlaythroughService.playthrough.issueCustomizationArgs = {
+        const currentPlaythrough = PlaythroughService.playthrough;
+        const issueCustomizationArgs = {
           state_name: {
             value:
               PlaythroughService.playthrough.actions[
                 PlaythroughService.playthrough.actions.length - 1
               ].actionCustomizationArgs.state_name.value
           },
-          time_spent_in_exp_in_secs: {
-            value: timeSpentInExpInSecs
+          time_spent_in_exp_in_msecs: {
+            value: timeSpentInExpInSecs * 1000
           }
         };
+
+        PlaythroughService.playthrough = (
+          this.playthroughObjectFactory
+            .createNewEarlyQuitPlaythrough(
+              currentPlaythrough.playthroughId, currentPlaythrough.expId,
+              currentPlaythrough.expVersion, issueCustomizationArgs,
+              currentPlaythrough.actions));
```

Comment:
@brianrodri ptal at the changes in this file, first.

Reply:
Can we exclude the changes to this file? This is all about to be heavily refactored anyway, this will only complicate the effort (see: #9497).

Reply:
The ts tests fail this way.

Reply:
OK, please isolate the changes which causes failure and leave it for a second PR which waits for #9497.

Reply:
Actually, I had to do this because I was declaring type in `PlaythroughObjectFactory` using discriminated unions. Do you mean that you will declare types in `playthroughObjectFactory` in #9497 ?